src/mip/extras/recording_connection.hpp

```diff
@@ -49,12 +49,25 @@ class RecordingConnection : public Connection
         return false;
     };
 
+    uint64_t recvFileBytesWritten()
+    {
+        return mRecvFileWritten;
+    }
+
+    uint64_t sendFileBytesWritten()
+    {
+        return mSendFileWritten;
+    }
+
 protected:
     Connection* mConnection;
 
     // Files may be NULL to not record one direction or the other
     std::ostream* mRecvFile;
     std::ostream* mSendFile;
+
+    uint64_t mRecvFileWritten = 0;
```

Comment:
I would change the types of these and the functions that return them to be `size_t`